kotest-assertions/kotest-assertions-core/src/commonMain/kotlin/io/kotest/matchers/ranges/bewithin.kt

```diff
@@ -0,0 +1,216 @@
+package io.kotest.matchers.ranges
+
+import io.kotest.assertions.print.print
+import io.kotest.matchers.Matcher
+import io.kotest.matchers.MatcherResult
+import io.kotest.matchers.should
+import io.kotest.matchers.shouldNot
+
+/**
+ * Verifies that this [ClosedRange] beWithin with another [ClosedRange].
+ *
+ * Assertion to check that this [ClosedRange] beWithin with another [ClosedRange].
```

Comment:
Are these AI generated? Feels like some more effort could be put into the wording here :)

* First sentence and 2nd sentence feels like they say the same thing with different words
* What does `beWithins` mean? Phrase it better and/or provide an example

Reply:
added examples